File: go/arrays/167_two_sum_II_input_array_is_sorted.go
package arrays

func TwoSum(numbers []int, target int) []int {

	left, right := 0, len(numbers)-1

	for left <= right {
		currSum := numbers[left] + numbers[right]
		if currSum == target {
			return []int{left + 1, right + 1}
		} else if currSum > target {
			right--
		} else {
			left++
		}
	}
	return []int{0, 0}
}

File: python/arrays/167_Two_Sum_II_Input_Array_Is_Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        s = 0
        e = len(numbers) - 1

        while s < e:
            currSum = numbers[s] + numbers[e]

            if currSum > target:
                e -= 1
            elif currSum < target:
                s += 1
            else:
                return [s+ 1, e+1]
